### Importing required libraries

In [1]:
import numpy as np
import pandas as pd

### Reading test cricket bowler data set

In [2]:
df = pd.read_csv("wickets.csv")

# Display of first 10 rows of the data set
display(df.head(10))

Player       Span   Mat  Inns  Balls   Runs  Wkts    BBI  \
0  M Muralitharan (ICC/SL)  1992-2010   133   230  44039  18180   800   9/51   
1           SK Warne (AUS)  1992-2007   145   273  40705  17995   708   8/71   
2        JM Anderson (ENG)  2003-2021  164*   304  35079  16575   623   7/42   
3         A Kumble (INDIA)  1990-2008   132   236  40850  18355   619  10/74   
4         GD McGrath (AUS)  1993-2007   124   243  29248  12186   563   8/24   
5          SCJ Broad (ENG)  2007-2021   149   274  29863  14590   524   8/15   
6            CA Walsh (WI)  1984-2001   132   242  30019  12688   519   7/37   
7            DW Steyn (SA)  2004-2019    93   171  18608  10077   439   7/51   
8      N Kapil Dev (INDIA)  1978-1994   131   227  27740  12867   434   9/83   
9        HMRKB Herath (SL)  1999-2018    93   170  25993  12157   433  9/127   

      BBM    Ave  Econ    SR   5  10  
0  16/220  22.72  2.47  55.0  67  22  
1  12/128  25.41  2.65  57.4  37  10  
2   11/71  26.60  2.83  56.3  30   3  
3  14/149  29.65  2.69  65.9  35   8  
4   10/27  21.64  2.49  51.9  29   3  
5  11/121  27.84  2.93  56.9  18   3  
6   13/55  24.44  2.53  57.8  22   3  
7   11/60  22.95  3.24  42.3  26   5  
8  11/146  29.64  2.78  63.9  23   2  
9  14/184  28.07  2.80  60.0  34   9

##### Description of each column of the data set
- Player: Name of the player and their country name or for whom they played for
- Span: Total duration of their career
- Mat: Total number of matches they played during their career
- Inns: Total number of innings they played
- Balls: Total number of balls they bowled
- Runs: Total number of runs they conceded from total number of balls bowled
- Wkts: Total number of wickets they took
- BBI: stands for 'Best Bowling in Innings' i.e. figure gives best bowling in one inning
- BBM: stands for 'Best Bowling in Match' and gives the combined score over 2 or more innings in one match
- Average: The average number of runs conceded per wicket. (Ave = Runs/W)
- Econ: The average number of runs conceded per over. (Econ = Runs/Overs bowled)
- SR: Stands for 'strike rate'. This is a bowling SR, which is different from batting SR. Bowling SR is defined for a bowler as the average number of balls bowled per wicket taken. The lower the strike rate, the more effective a bowler is at taking wickets quickly.
- 5: The number of innings in which the bowler took at least five wickets
- 10: The number of matches in which the bowler took at least ten wickets


###  number of rows and columns of the dataframe

In [3]:
# number of rows
print("number of rows = ", df.shape[0])

# number of columns
print("number of columns = ", df.shape[1])

number of rows =  79
number of columns =  14


###  Data statistics and data types check

In [4]:
# checking data statistics
display(df.describe())

# checking for missing values and data types of each column
print(df.info())

Inns         Balls          Runs        Wkts        Ave  \
count   79.000000     79.000000     79.000000   79.000000  79.000000   
mean   144.911392  18638.354430   8599.354430  317.215190  27.469747   
std     51.180222   7199.256972   3085.168807  121.924911   3.655658   
min     67.000000   8785.000000   4846.000000  200.000000  20.940000   
25%    110.000000  13583.000000   6456.500000  229.000000  24.500000   
50%    129.000000  16498.000000   7742.000000  266.000000  28.000000   
75%    169.000000  21742.500000   9756.000000  374.500000  29.870000   
max    304.000000  44039.000000  18355.000000  800.000000  34.790000   

            Econ         SR          5         10  
count  79.000000  79.000000  79.000000  79.000000  
mean    2.806835  59.193671  16.354430   2.797468  
std     0.351577   9.350132   9.642372   3.235935  
min     1.980000  41.200000   3.000000   0.000000  
25%     2.600000  53.300000   9.500000   1.000000  
50%     2.820000  57.400000  14.000000   2.000000  
75%     3.080000  63.950000  20.500000   3.500000  
max     3.460000  91.900000  67.000000  22.000000

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 79 entries, 0 to 78
Data columns (total 14 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Player  79 non-null     object 
 1   Span    79 non-null     object 
 2   Mat     79 non-null     object 
 3   Inns    79 non-null     int64  
 4   Balls   79 non-null     int64  
 5   Runs    79 non-null     int64  
 6   Wkts    79 non-null     int64  
 7   BBI     79 non-null     object 
 8   BBM     79 non-null     object 
 9   Ave     79 non-null     float64
 10  Econ    79 non-null     float64
 11  SR      79 non-null     float64
 12  5       79 non-null     int64  
 13  10      79 non-null     int64  
dtypes: float64(3), int64(6), object(5)
memory usage: 8.8+ KB
None


#### There is no missing value found in the data set

### Renaming the column names of the data set

In [5]:
# column names
print(df.columns)

Index(['Player', 'Span', 'Mat', 'Inns', 'Balls', 'Runs', 'Wkts', 'BBI', 'BBM',
       'Ave', 'Econ', 'SR', '5', '10'],
      dtype='object')


In [6]:
df = df.rename(columns={'Mat':'Match', 
                        'Inns':'Innings',
                        'Wkts': 'Wickets',
                        'BBI': 'Best_bowling_innings',
                        'BBM': 'Best_bowling_match',
                        'Ave': 'Average',
                        'Econ': 'Economy',
                        'SR': 'Bowling_strike_rate',
                        5: 'Five_wickets_taken',
                        10: 'Ten_wickets_taken',})

display(df.head())

Player       Span Match  Innings  Balls   Runs  Wickets  \
0  M Muralitharan (ICC/SL)  1992-2010   133      230  44039  18180      800   
1           SK Warne (AUS)  1992-2007   145      273  40705  17995      708   
2        JM Anderson (ENG)  2003-2021  164*      304  35079  16575      623   
3         A Kumble (INDIA)  1990-2008   132      236  40850  18355      619   
4         GD McGrath (AUS)  1993-2007   124      243  29248  12186      563   

  Best_bowling_innings Best_bowling_match  Average  Economy  \
0                 9/51             16/220    22.72     2.47   
1                 8/71             12/128    25.41     2.65   
2                 7/42              11/71    26.60     2.83   
3                10/74             14/149    29.65     2.69   
4                 8/24              10/27    21.64     2.49   

   Bowling_strike_rate   5  10  
0                 55.0  67  22  
1                 57.4  37  10  
2                 56.3  30   3  
3                 65.9  35   8  
4                 51.9  29   3

### Romoving columns 

In [36]:
# Remove two columns name is 'Best_bowling_innings' and 'Best_bowling_match'
df.drop(['Best_bowling_innings', 'Best_bowling_match'], axis = 1)

Player       Span Match  Innings  Balls   Runs  Wickets  \
0   M Muralitharan (ICC/SL)  1992-2010   133      230  44039  18180      800   
1            SK Warne (AUS)  1992-2007   145      273  40705  17995      708   
2         JM Anderson (ENG)  2003-2021  164*      304  35079  16575      623   
3          A Kumble (INDIA)  1990-2008   132      236  40850  18355      619   
4          GD McGrath (AUS)  1993-2007   124      243  29248  12186      563   
..                      ...        ...   ...      ...    ...    ...      ...   
74        SCG MacGill (AUS)  1998-2008    44       85  11237   6038      208   
75    Saqlain Mushtaq (PAK)  1995-2004    49       86  14070   6206      208   
76         AME Roberts (WI)  1974-1983    47       90  11135   5174      202   
77            JA Snow (ENG)  1965-1976    49       93  12021   5387      202   
78         JR Thomson (AUS)  1972-1985    51       90  10535   5601      200   

    Average  Economy  Bowling_strike_rate   5  10  
0     22.72     2.47                 55.0  67  22  
1     25.41     2.65                 57.4  37  10  
2     26.60     2.83                 56.3  30   3  
3     29.65     2.69                 65.9  35   8  
4     21.64     2.49                 51.9  29   3  
..      ...      ...                  ...  ..  ..  
74    29.02     3.22                 54.0  12   2  
75    29.83     2.64                 67.6  13   3  
76    25.61     2.78                 55.1  11   2  
77    26.66     2.68                 59.5   8   1  
78    28.00     3.18                 52.6   8   0  

[79 rows x 12 columns]

In [37]:
# splitting the 'Player' column to get the information about 'Country'
df_player = df['Player'].str.split("(", expand=True)

display(df_player.head(5))

0        1
0  M Muralitharan   ICC/SL)
1        SK Warne      AUS)
2     JM Anderson      ENG)
3        A Kumble    INDIA)
4      GD McGrath      AUS)

In [38]:
# concatinating 'Country' with the main dataframe
df = pd.concat([df, df_player], axis=1)

display(df.head())

Player       Span Match  Innings  Balls   Runs  Wickets  \
0  M Muralitharan (ICC/SL)  1992-2010   133      230  44039  18180      800   
1           SK Warne (AUS)  1992-2007   145      273  40705  17995      708   
2        JM Anderson (ENG)  2003-2021  164*      304  35079  16575      623   
3         A Kumble (INDIA)  1990-2008   132      236  40850  18355      619   
4         GD McGrath (AUS)  1993-2007   124      243  29248  12186      563   

  Best_bowling_innings Best_bowling_match  Average  Economy  \
0                 9/51             16/220    22.72     2.47   
1                 8/71             12/128    25.41     2.65   
2                 7/42              11/71    26.60     2.83   
3                10/74             14/149    29.65     2.69   
4                 8/24              10/27    21.64     2.49   

   Bowling_strike_rate   5  10                0        1  
0                 55.0  67  22  M Muralitharan   ICC/SL)  
1                 57.4  37  10        SK Warne      AUS)  
2                 56.3  30   3     JM Anderson      ENG)  
3                 65.9  35   8        A Kumble    INDIA)  
4                 51.9  29   3      GD McGrath      AUS)

In [39]:
# dropping the 'Player' column
df = df.drop('Player', axis=1)

display(df.head())

Span Match  Innings  Balls   Runs  Wickets Best_bowling_innings  \
0  1992-2010   133      230  44039  18180      800                 9/51   
1  1992-2007   145      273  40705  17995      708                 8/71   
2  2003-2021  164*      304  35079  16575      623                 7/42   
3  1990-2008   132      236  40850  18355      619                10/74   
4  1993-2007   124      243  29248  12186      563                 8/24   

  Best_bowling_match  Average  Economy  Bowling_strike_rate   5  10  \
0             16/220    22.72     2.47                 55.0  67  22   
1             12/128    25.41     2.65                 57.4  37  10   
2              11/71    26.60     2.83                 56.3  30   3   
3             14/149    29.65     2.69                 65.9  35   8   
4              10/27    21.64     2.49                 51.9  29   3   

                 0        1  
0  M Muralitharan   ICC/SL)  
1        SK Warne      AUS)  
2     JM Anderson      ENG)  
3        A Kumble    INDIA)  
4      GD McGrath      AUS)

In [40]:
# renaming the column names
df = df.rename(columns={0: 'Player',
                        1: 'Country'})

display(df.head())

Span Match  Innings  Balls   Runs  Wickets Best_bowling_innings  \
0  1992-2010   133      230  44039  18180      800                 9/51   
1  1992-2007   145      273  40705  17995      708                 8/71   
2  2003-2021  164*      304  35079  16575      623                 7/42   
3  1990-2008   132      236  40850  18355      619                10/74   
4  1993-2007   124      243  29248  12186      563                 8/24   

  Best_bowling_match  Average  Economy  Bowling_strike_rate   5  10  \
0             16/220    22.72     2.47                 55.0  67  22   
1             12/128    25.41     2.65                 57.4  37  10   
2              11/71    26.60     2.83                 56.3  30   3   
3             14/149    29.65     2.69                 65.9  35   8   
4              10/27    21.64     2.49                 51.9  29   3   

            Player  Country  
0  M Muralitharan   ICC/SL)  
1        SK Warne      AUS)  
2     JM Anderson      ENG)  
3        A Kumble    INDIA)  
4      GD McGrath      AUS)

In [41]:
# remove the ")" from the 'Country' column
df['Country'] = df['Country'].str.replace(")", "")

display(df.head())

Span Match  Innings  Balls   Runs  Wickets Best_bowling_innings  \
0  1992-2010   133      230  44039  18180      800                 9/51   
1  1992-2007   145      273  40705  17995      708                 8/71   
2  2003-2021  164*      304  35079  16575      623                 7/42   
3  1990-2008   132      236  40850  18355      619                10/74   
4  1993-2007   124      243  29248  12186      563                 8/24   

  Best_bowling_match  Average  Economy  Bowling_strike_rate   5  10  \
0             16/220    22.72     2.47                 55.0  67  22   
1             12/128    25.41     2.65                 57.4  37  10   
2              11/71    26.60     2.83                 56.3  30   3   
3             14/149    29.65     2.69                 65.9  35   8   
4              10/27    21.64     2.49                 51.9  29   3   

            Player Country  
0  M Muralitharan   ICC/SL  
1        SK Warne      AUS  
2     JM Anderson      ENG  
3        A Kumble    INDIA  
4      GD McGrath      AUS

In [42]:
print(df.columns)

Index(['Span', 'Match', 'Innings', 'Balls', 'Runs', 'Wickets',
       'Best_bowling_innings', 'Best_bowling_match', 'Average', 'Economy',
       'Bowling_strike_rate', '5', '10', 'Player', 'Country'],
      dtype='object')


In [44]:
# rearrange the columns
new_col_sequence = ['Player', 'Country','Span', 'Match', 'Innings', 'Balls', 'Runs', 'Wickets',
       'Best_bowling_innings', 'Best_bowling_match', 'Average', 'Economy',
       'Bowling_strike_rate', '5', '10']

In [45]:
df = df[new_col_sequence]

display(df.head())

Player Country       Span Match  Innings  Balls   Runs  Wickets  \
0  M Muralitharan   ICC/SL  1992-2010   133      230  44039  18180      800   
1        SK Warne      AUS  1992-2007   145      273  40705  17995      708   
2     JM Anderson      ENG  2003-2021  164*      304  35079  16575      623   
3        A Kumble    INDIA  1990-2008   132      236  40850  18355      619   
4      GD McGrath      AUS  1993-2007   124      243  29248  12186      563   

  Best_bowling_innings Best_bowling_match  Average  Economy  \
0                 9/51             16/220    22.72     2.47   
1                 8/71             12/128    25.41     2.65   
2                 7/42              11/71    26.60     2.83   
3                10/74             14/149    29.65     2.69   
4                 8/24              10/27    21.64     2.49   

   Bowling_strike_rate   5  10  
0                 55.0  67  22  
1                 57.4  37  10  
2                 56.3  30   3  
3                 65.9  35   8  
4                 51.9  29   3

### creation a column based on a condition or function

In [46]:
def icc_check(x):
    if "ICC" in x:
        return "Yes"
    else:
        return "No"

In [47]:
# Checking whether a player had played for ICC or not
df['played_for_ICC'] = df['Country'].apply(icc_check)

display(df.head(10))

Player Country       Span Match  Innings  Balls   Runs  Wickets  \
0  M Muralitharan   ICC/SL  1992-2010   133      230  44039  18180      800   
1        SK Warne      AUS  1992-2007   145      273  40705  17995      708   
2     JM Anderson      ENG  2003-2021  164*      304  35079  16575      623   
3        A Kumble    INDIA  1990-2008   132      236  40850  18355      619   
4      GD McGrath      AUS  1993-2007   124      243  29248  12186      563   
5       SCJ Broad      ENG  2007-2021   149      274  29863  14590      524   
6        CA Walsh       WI  1984-2001   132      242  30019  12688      519   
7        DW Steyn       SA  2004-2019    93      171  18608  10077      439   
8     N Kapil Dev    INDIA  1978-1994   131      227  27740  12867      434   
9    HMRKB Herath       SL  1999-2018    93      170  25993  12157      433   

  Best_bowling_innings Best_bowling_match  Average  Economy  \
0                 9/51             16/220    22.72     2.47   
1                 8/71             12/128    25.41     2.65   
2                 7/42              11/71    26.60     2.83   
3                10/74             14/149    29.65     2.69   
4                 8/24              10/27    21.64     2.49   
5                 8/15             11/121    27.84     2.93   
6                 7/37              13/55    24.44     2.53   
7                 7/51              11/60    22.95     3.24   
8                 9/83             11/146    29.64     2.78   
9                9/127             14/184    28.07     2.80   

   Bowling_strike_rate   5  10 played_for_ICC  
0                 55.0  67  22            Yes  
1                 57.4  37  10             No  
2                 56.3  30   3             No  
3                 65.9  35   8             No  
4                 51.9  29   3             No  
5                 56.9  18   3             No  
6                 57.8  22   3             No  
7                 42.3  26   5             No  
8                 63.9  23   2             No  
9                 60.0  34   9             No

In [48]:
# Displaying how many players played for ICC
df['played_for_ICC'].value_counts()

No     74
Yes     5
Name: played_for_ICC, dtype: int64

In [49]:
# Removing 'ICC/' from the data set to make the country name free
df['Country'] = df['Country'].str.replace("ICC/", "")

display(df.head())

Player Country       Span Match  Innings  Balls   Runs  Wickets  \
0  M Muralitharan       SL  1992-2010   133      230  44039  18180      800   
1        SK Warne      AUS  1992-2007   145      273  40705  17995      708   
2     JM Anderson      ENG  2003-2021  164*      304  35079  16575      623   
3        A Kumble    INDIA  1990-2008   132      236  40850  18355      619   
4      GD McGrath      AUS  1993-2007   124      243  29248  12186      563   

  Best_bowling_innings Best_bowling_match  Average  Economy  \
0                 9/51             16/220    22.72     2.47   
1                 8/71             12/128    25.41     2.65   
2                 7/42              11/71    26.60     2.83   
3                10/74             14/149    29.65     2.69   
4                 8/24              10/27    21.64     2.49   

   Bowling_strike_rate   5  10 played_for_ICC  
0                 55.0  67  22            Yes  
1                 57.4  37  10             No  
2                 56.3  30   3             No  
3                 65.9  35   8             No  
4                 51.9  29   3             No

In [50]:
df['Country'].value_counts()

AUS        18
ENG        13
INDIA      10
WI          9
SA          8
PAK         7
NZ          7
SL          3
ENG/ICC     2
BDESH       1
ZIM         1
Name: Country, dtype: int64

In [51]:
# Now removing '/ICC' from the data set to make the country name free
df['Country'] = df['Country'].str.replace("/ICC", "")

display(df.head())

Player Country       Span Match  Innings  Balls   Runs  Wickets  \
0  M Muralitharan       SL  1992-2010   133      230  44039  18180      800   
1        SK Warne      AUS  1992-2007   145      273  40705  17995      708   
2     JM Anderson      ENG  2003-2021  164*      304  35079  16575      623   
3        A Kumble    INDIA  1990-2008   132      236  40850  18355      619   
4      GD McGrath      AUS  1993-2007   124      243  29248  12186      563   

  Best_bowling_innings Best_bowling_match  Average  Economy  \
0                 9/51             16/220    22.72     2.47   
1                 8/71             12/128    25.41     2.65   
2                 7/42              11/71    26.60     2.83   
3                10/74             14/149    29.65     2.69   
4                 8/24              10/27    21.64     2.49   

   Bowling_strike_rate   5  10 played_for_ICC  
0                 55.0  67  22            Yes  
1                 57.4  37  10             No  
2                 56.3  30   3             No  
3                 65.9  35   8             No  
4                 51.9  29   3             No

In [52]:
df['Country'].value_counts()

AUS      18
ENG      15
INDIA    10
WI        9
SA        8
PAK       7
NZ        7
SL        3
BDESH     1
ZIM       1
Name: Country, dtype: int64

In [53]:
# splitting the 'Span' column based on the "-"
df_span = df['Span'].str.split("-", expand=True)
display(df.head(5))

Player Country       Span Match  Innings  Balls   Runs  Wickets  \
0  M Muralitharan       SL  1992-2010   133      230  44039  18180      800   
1        SK Warne      AUS  1992-2007   145      273  40705  17995      708   
2     JM Anderson      ENG  2003-2021  164*      304  35079  16575      623   
3        A Kumble    INDIA  1990-2008   132      236  40850  18355      619   
4      GD McGrath      AUS  1993-2007   124      243  29248  12186      563   

  Best_bowling_innings Best_bowling_match  Average  Economy  \
0                 9/51             16/220    22.72     2.47   
1                 8/71             12/128    25.41     2.65   
2                 7/42              11/71    26.60     2.83   
3                10/74             14/149    29.65     2.69   
4                 8/24              10/27    21.64     2.49   

   Bowling_strike_rate   5  10 played_for_ICC  
0                 55.0  67  22            Yes  
1                 57.4  37  10             No  
2                 56.3  30   3             No  
3                 65.9  35   8             No  
4                 51.9  29   3             No

In [54]:
# concatinating the new dataframe with the main dataframe
df = pd.concat([df, df_span], axis=1)
display(df.head(5))

Player Country       Span Match  Innings  Balls   Runs  Wickets  \
0  M Muralitharan       SL  1992-2010   133      230  44039  18180      800   
1        SK Warne      AUS  1992-2007   145      273  40705  17995      708   
2     JM Anderson      ENG  2003-2021  164*      304  35079  16575      623   
3        A Kumble    INDIA  1990-2008   132      236  40850  18355      619   
4      GD McGrath      AUS  1993-2007   124      243  29248  12186      563   

  Best_bowling_innings Best_bowling_match  Average  Economy  \
0                 9/51             16/220    22.72     2.47   
1                 8/71             12/128    25.41     2.65   
2                 7/42              11/71    26.60     2.83   
3                10/74             14/149    29.65     2.69   
4                 8/24              10/27    21.64     2.49   

   Bowling_strike_rate   5  10 played_for_ICC     0     1  
0                 55.0  67  22            Yes  1992  2010  
1                 57.4  37  10             No  1992  2007  
2                 56.3  30   3             No  2003  2021  
3                 65.9  35   8             No  1990  2008  
4                 51.9  29   3             No  1993  2007

In [55]:
# renaming the newly created column names
df = df.rename(columns={0: "start_year",
                        1: "end_year"})
display(df.head(5))

Player Country       Span Match  Innings  Balls   Runs  Wickets  \
0  M Muralitharan       SL  1992-2010   133      230  44039  18180      800   
1        SK Warne      AUS  1992-2007   145      273  40705  17995      708   
2     JM Anderson      ENG  2003-2021  164*      304  35079  16575      623   
3        A Kumble    INDIA  1990-2008   132      236  40850  18355      619   
4      GD McGrath      AUS  1993-2007   124      243  29248  12186      563   

  Best_bowling_innings Best_bowling_match  Average  Economy  \
0                 9/51             16/220    22.72     2.47   
1                 8/71             12/128    25.41     2.65   
2                 7/42              11/71    26.60     2.83   
3                10/74             14/149    29.65     2.69   
4                 8/24              10/27    21.64     2.49   

   Bowling_strike_rate   5  10 played_for_ICC start_year end_year  
0                 55.0  67  22            Yes       1992     2010  
1                 57.4  37  10             No       1992     2007  
2                 56.3  30   3             No       2003     2021  
3                 65.9  35   8             No       1990     2008  
4                 51.9  29   3             No       1993     2007

In [56]:
# removing the "Span" column
df = df.drop("Span", axis=1)

display(df.head())

Player Country Match  Innings  Balls   Runs  Wickets  \
0  M Muralitharan       SL   133      230  44039  18180      800   
1        SK Warne      AUS   145      273  40705  17995      708   
2     JM Anderson      ENG  164*      304  35079  16575      623   
3        A Kumble    INDIA   132      236  40850  18355      619   
4      GD McGrath      AUS   124      243  29248  12186      563   

  Best_bowling_innings Best_bowling_match  Average  Economy  \
0                 9/51             16/220    22.72     2.47   
1                 8/71             12/128    25.41     2.65   
2                 7/42              11/71    26.60     2.83   
3                10/74             14/149    29.65     2.69   
4                 8/24              10/27    21.64     2.49   

   Bowling_strike_rate   5  10 played_for_ICC start_year end_year  
0                 55.0  67  22            Yes       1992     2010  
1                 57.4  37  10             No       1992     2007  
2                 56.3  30   3             No       2003     2021  
3                 65.9  35   8             No       1990     2008  
4                 51.9  29   3             No       1993     2007

In [57]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 79 entries, 0 to 78
Data columns (total 17 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Player                79 non-null     object 
 1   Country               79 non-null     object 
 2   Match                 79 non-null     object 
 3   Innings               79 non-null     int64  
 4   Balls                 79 non-null     int64  
 5   Runs                  79 non-null     int64  
 6   Wickets               79 non-null     int64  
 7   Best_bowling_innings  79 non-null     object 
 8   Best_bowling_match    79 non-null     object 
 9   Average               79 non-null     float64
 10  Economy               79 non-null     float64
 11  Bowling_strike_rate   79 non-null     float64
 12  5                     79 non-null     int64  
 13  10                    79 non-null     int64  
 14  played_for_ICC        79 non-null     object 
 15  start_year            79 

In [58]:
df['start_year'] = df['start_year'].astype('int') 
df['end_year'] = df['end_year'].astype('int')

print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 79 entries, 0 to 78
Data columns (total 17 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Player                79 non-null     object 
 1   Country               79 non-null     object 
 2   Match                 79 non-null     object 
 3   Innings               79 non-null     int64  
 4   Balls                 79 non-null     int64  
 5   Runs                  79 non-null     int64  
 6   Wickets               79 non-null     int64  
 7   Best_bowling_innings  79 non-null     object 
 8   Best_bowling_match    79 non-null     object 
 9   Average               79 non-null     float64
 10  Economy               79 non-null     float64
 11  Bowling_strike_rate   79 non-null     float64
 12  5                     79 non-null     int64  
 13  10                    79 non-null     int64  
 14  played_for_ICC        79 non-null     object 
 15  start_year            79 

In [59]:
df['years_played'] = df['end_year'] - df['start_year']

df = df.drop(['start_year', "end_year"], axis=1)

display(df.head(10))

Player Country Match  Innings  Balls   Runs  Wickets  \
0  M Muralitharan       SL   133      230  44039  18180      800   
1        SK Warne      AUS   145      273  40705  17995      708   
2     JM Anderson      ENG  164*      304  35079  16575      623   
3        A Kumble    INDIA   132      236  40850  18355      619   
4      GD McGrath      AUS   124      243  29248  12186      563   
5       SCJ Broad      ENG   149      274  29863  14590      524   
6        CA Walsh       WI   132      242  30019  12688      519   
7        DW Steyn       SA    93      171  18608  10077      439   
8     N Kapil Dev    INDIA   131      227  27740  12867      434   
9    HMRKB Herath       SL    93      170  25993  12157      433   

  Best_bowling_innings Best_bowling_match  Average  Economy  \
0                 9/51             16/220    22.72     2.47   
1                 8/71             12/128    25.41     2.65   
2                 7/42              11/71    26.60     2.83   
3                10/74             14/149    29.65     2.69   
4                 8/24              10/27    21.64     2.49   
5                 8/15             11/121    27.84     2.93   
6                 7/37              13/55    24.44     2.53   
7                 7/51              11/60    22.95     3.24   
8                 9/83             11/146    29.64     2.78   
9                9/127             14/184    28.07     2.80   

   Bowling_strike_rate   5  10 played_for_ICC  years_played  
0                 55.0  67  22            Yes            18  
1                 57.4  37  10             No            15  
2                 56.3  30   3             No            18  
3                 65.9  35   8             No            18  
4                 51.9  29   3             No            14  
5                 56.9  18   3             No            14  
6                 57.8  22   3             No            17  
7                 42.3  26   5             No            15  
8                 63.9  23   2             No            16  
9                 60.0  34   9             No            19

In [60]:
df.sort_values(by='years_played', ascending = False).head(5)

Player Country Match  Innings  Balls   Runs  Wickets  \
21      Imran Khan      PAK    88      142  19458   8258      362   
55       GS Sobers       WI    93      159  21599   7999      235   
9     HMRKB Herath       SL    93      170  25993  12157      433   
0   M Muralitharan       SL   133      230  44039  18180      800   
2      JM Anderson      ENG  164*      304  35079  16575      623   

   Best_bowling_innings Best_bowling_match  Average  Economy  \
21                 8/58             14/116    22.81     2.54   
55                 6/73               8/80    34.03     2.22   
9                 9/127             14/184    28.07     2.80   
0                  9/51             16/220    22.72     2.47   
2                  7/42              11/71    26.60     2.83   

    Bowling_strike_rate   5  10 played_for_ICC  years_played  
21                 53.7  23   6             No            21  
55                 91.9   6   0             No            20  
9                  60.0  34   9             No            19  
0                  55.0  67  22            Yes            18  
2                  56.3  30   3             No            18

### Imran Khan had played for the longest period of time (21 years)

In [61]:
df.sort_values(by='years_played', ascending = False).tail(5)

Player Country Match  Innings  Balls  Runs  Wickets  \
54    Yasir Shah      PAK   46*       84  13607  7248      235   
61   SJ Harmison      ENG    63      115  13375  7192      226   
72  JR Hazlewood      AUS    55      103  11887  5438      212   
71      K Rabada       SA    47       86   8785  4846      213   
44      GP Swann      ENG    60      109  15349  7642      255   

   Best_bowling_innings Best_bowling_match  Average  Economy  \
54                 8/41             14/184    30.84     3.19   
61                 7/12              11/76    31.82     3.22   
72                 6/67              9/115    25.65     2.74   
71                7/112             13/144    22.75     3.30   
44                 6/65             10/132    29.96     2.98   

    Bowling_strike_rate   5  10 played_for_ICC  years_played  
54                 57.9  16   3             No             7  
61                 59.1   8   1            Yes             7  
72                 56.0   9   0             No             7  
71                 41.2  10   4             No             6  
44                 60.1  17   3             No             5

### GP Swann had played for the shortest period of time (5 years)

In [62]:
# Number of Australian Bowlers present in the dataset
df['Country'].value_counts()

AUS      18
ENG      15
INDIA    10
WI        9
SA        8
PAK       7
NZ        7
SL        3
BDESH     1
ZIM       1
Name: Country, dtype: int64

#### There are 18 Australian bowlers in the dataset
#### There is only one bowler from Bangladesh 

### Which player had the lowest economy rate?

In [63]:
df.sort_values(by='Economy', ascending = False).tail(5)

Player Country Match  Innings  Balls  Runs  Wickets  \
68   CV Grimmett      AUS    37       67  14513  5231      216   
39       BS Bedi    INDIA    67      118  21364  7637      266   
47      R Benaud      AUS    63      116  19108  6704      248   
35  DL Underwood      ENG    86      151  21862  7674      297   
32      LR Gibbs       WI    79      148  27115  8989      309   

   Best_bowling_innings Best_bowling_match  Average  Economy  \
68                 7/40             14/199    24.21     2.16   
39                 7/98             10/194    28.71     2.14   
47                 7/72             11/105    27.03     2.10   
35                 8/51              13/71    25.83     2.10   
32                 8/38             11/157    29.09     1.98   

    Bowling_strike_rate   5  10 played_for_ICC  years_played  
68                 67.1  21   7             No            11  
39                 80.3  14   1             No            13  
47                 77.0  16   1             No            12  
35                 73.6  17   6             No            16  
32                 87.7  18   2             No            18

#### LR Gibbs had the lowest economy rate (1.98)

### Which player had the lowest strike rate?

In [64]:
df.sort_values(by='Bowling_strike_rate', ascending = False).tail(5)

Player Country Match  Innings  Balls   Runs  Wickets  \
25     AA Donald       SA    72      129  15519   7344      330   
19   MD Marshall       WI    81      151  17584   7876      376   
20  Waqar Younis      PAK    87      154  16224   8788      373   
7       DW Steyn       SA    93      171  18608  10077      439   
71      K Rabada       SA    47       86   8785   4846      213   

   Best_bowling_innings Best_bowling_match  Average  Economy  \
25                 8/71             12/139    22.25     2.83   
19                 7/22              11/89    20.94     2.68   
20                 7/76             13/135    23.56     3.25   
7                  7/51              11/60    22.95     3.24   
71                7/112             13/144    22.75     3.30   

    Bowling_strike_rate   5  10 played_for_ICC  years_played  
25                 47.0  20   3             No            10  
19                 46.7  22   4             No            13  
20                 43.4  22   5             No            14  
7                  42.3  26   5             No            15  
71                 41.2  10   4             No             6

#### K Rabada had the lowest strike rate (41.2)

### Which player had the lowest bowling average?

In [65]:
df.sort_values(by='Average', ascending = False).tail(5)

Player Country Match  Innings  Balls   Runs  Wickets  \
4    GD McGrath      AUS   124      243  29248  12186      563   
33   FS Trueman      ENG    67      127  15178   6625      307   
15  CEL Ambrose       WI    98      179  22103   8501      405   
41     J Garner       WI    58      111  13169   5433      259   
19  MD Marshall       WI    81      151  17584   7876      376   

   Best_bowling_innings Best_bowling_match  Average  Economy  \
4                  8/24              10/27    21.64     2.49   
33                 8/31             12/119    21.57     2.61   
15                 8/45              11/84    20.99     2.30   
41                 6/56              9/108    20.97     2.47   
19                 7/22              11/89    20.94     2.68   

    Bowling_strike_rate   5  10 played_for_ICC  years_played  
4                  51.9  29   3             No            14  
33                 49.4  17   3             No            13  
15                 54.5  22   3             No            12  
41                 50.8   7   0             No            10  
19                 46.7  22   4             No            13

#### MD Marshall had the lowest bowling average (20.94)